macros.in

```diff
@@ -1185,6 +1205,24 @@ package or when debugging this package.\
 #------------------------------------------------------------------------------
 # Macros for further automated spec %setup and patch application
 
+# Build the scm-indepdendent default apply flags
+%_default_apply_flags %{expand:%%{?_default_%{__scm}_apply_flags}}
+
+# Choose between default apply flags and user-defined apply flags
+%_apply_flags %{expand:%%{!?__scm_%{__scm}_apply_flags:%%{?_default_apply_flags}}%%{?__scm_%{__scm}_apply_flags:%%{?__scm_%{__scm}_apply_flags}}}
+
+# Build the scm-indepdendent default import flags
+%_default_import_flags %{expand:%%{?_default_%{__scm}_import_flags}}
+
+# Choose between default import flags and user-defined import flags
+%_import_flags %{expand:%%{!?__scm_%{__scm}_import_flags:%%{?_default_import_flags}}%%{?__scm_%{__scm}_import_flags:%%{?__scm_%{__scm}_import_flags}}}
+
+# Build the scm-independant default commit flags
+%_default_commit_flags %{expand:%%{?_default_%{__scm}_commit_flags}}
+
+# Choose between default commit flags and user-defined commit flags
+%_commit_flags %{expand:%%{!?__scm_%{__scm}_commit_flags:%%{?_default_commit_flags}}%%{?__scm_%{__scm}_commit_flags:%%{?__scm_%{__scm}_commit_flags}}}
```

Comment:
%_apply_flags, %_import_flags and %_commit flags are awfully generic names. Please prepend "scm" or such in there someplace, eg %_scm_apply_flags (but I didn't verify whether that clashes with something)